2023-07-03 11:54:14.785 - INFO  [           main] o.s.b.StartupInfoLogger        : Starting EmployeeApplication using Java 1.8.0_342 on LAPTOP-T5ICR3J1 with PID 19636 (D:\Employee_Management\target\classes started by 48518 in D:\Employee_Management)
2023-07-03 11:54:14.802 - INFO  [           main] o.s.b.SpringApplication        : No active profile set, falling back to 1 default profile: "default"
2023-07-03 11:54:15.690 - INFO  [           main] epositoryConfigurationDelegate : Multiple Spring Data modules found, entering strict repository configuration mode
2023-07-03 11:54:15.695 - INFO  [           main] epositoryConfigurationDelegate : Bootstrapping Spring Data Redis repositories in DEFAULT mode.
2023-07-03 11:54:15.738 - INFO  [           main] epositoryConfigurationDelegate : Finished Spring Data repository scanning in 12 ms. Found 0 Redis repository interfaces.
2023-07-03 11:54:16.338 - INFO  [           main] o.s.b.w.e.t.TomcatWebServer    : Tomcat initialized with port(s): 8080 (http)
2023-07-03 11:54:16.345 - INFO  [           main] o.a.j.l.DirectJDKLog           : Initializing ProtocolHandler ["http-nio-8080"]
2023-07-03 11:54:16.347 - INFO  [           main] o.a.j.l.DirectJDKLog           : Starting service [Tomcat]
2023-07-03 11:54:16.347 - INFO  [           main] o.a.j.l.DirectJDKLog           : Starting Servlet engine: [Apache Tomcat/9.0.76]
2023-07-03 11:54:16.495 - INFO  [           main] o.a.j.l.DirectJDKLog           : Initializing Spring embedded WebApplicationContext
2023-07-03 11:54:16.496 - INFO  [           main] letWebServerApplicationContext : Root WebApplicationContext: initialization completed in 1645 ms
2023-07-03 11:54:17.488 - INFO  [           main] edRequestMappingHandlerMapping : Mapped URL path [/v2/api-docs] onto method [springfox.documentation.swagger2.web.Swagger2Controller#getDocumentation(String, HttpServletRequest)]
2023-07-03 11:54:20.635 - WARN  [           main] c.s.AbstractApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'enableRedisKeyspaceNotificationsInitializer' defined in class path resource [org/springframework/boot/autoconfigure/session/RedisSessionConfiguration$SpringBootRedisHttpSessionConfiguration.class]: Invocation of init method failed; nested exception is org.springframework.data.redis.RedisConnectionFailureException: Unable to connect to Redis; nested exception is io.lettuce.core.RedisConnectionException: Unable to connect to localhost:6379
2023-07-03 11:54:20.798 - INFO  [           main] o.a.j.l.DirectJDKLog           : Stopping service [Tomcat]
2023-07-03 11:54:20.808 - INFO  [           main] valuationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2023-07-03 11:55:04.287 - INFO  [           main] o.s.b.StartupInfoLogger        : Starting EmployeeApplication using Java 1.8.0_342 on LAPTOP-T5ICR3J1 with PID 12004 (D:\Employee_Management\target\classes started by 48518 in D:\Employee_Management)
2023-07-03 11:55:04.298 - INFO  [           main] o.s.b.SpringApplication        : No active profile set, falling back to 1 default profile: "default"
2023-07-03 11:55:05.152 - INFO  [           main] epositoryConfigurationDelegate : Multiple Spring Data modules found, entering strict repository configuration mode
2023-07-03 11:55:05.154 - INFO  [           main] epositoryConfigurationDelegate : Bootstrapping Spring Data Redis repositories in DEFAULT mode.
2023-07-03 11:55:05.178 - INFO  [           main] epositoryConfigurationDelegate : Finished Spring Data repository scanning in 8 ms. Found 0 Redis repository interfaces.
2023-07-03 11:55:05.764 - INFO  [           main] o.s.b.w.e.t.TomcatWebServer    : Tomcat initialized with port(s): 8080 (http)
2023-07-03 11:55:05.771 - INFO  [           main] o.a.j.l.DirectJDKLog           : Initializing ProtocolHandler ["http-nio-8080"]
2023-07-03 11:55:05.773 - INFO  [           main] o.a.j.l.DirectJDKLog           : Starting service [Tomcat]
2023-07-03 11:55:05.773 - INFO  [           main] o.a.j.l.DirectJDKLog           : Starting Servlet engine: [Apache Tomcat/9.0.76]
2023-07-03 11:55:05.970 - INFO  [           main] o.a.j.l.DirectJDKLog           : Initializing Spring embedded WebApplicationContext
2023-07-03 11:55:05.970 - INFO  [           main] letWebServerApplicationContext : Root WebApplicationContext: initialization completed in 1620 ms
2023-07-03 11:55:06.908 - INFO  [           main] edRequestMappingHandlerMapping : Mapped URL path [/v2/api-docs] onto method [springfox.documentation.swagger2.web.Swagger2Controller#getDocumentation(String, HttpServletRequest)]
2023-07-03 11:55:08.113 - INFO  [           main] o.a.j.l.DirectJDKLog           : Starting ProtocolHandler ["http-nio-8080"]
2023-07-03 11:55:08.124 - INFO  [           main] o.s.b.w.e.t.TomcatWebServer    : Tomcat started on port(s): 8080 (http) with context path ''
2023-07-03 11:55:08.125 - INFO  [           main] cumentationPluginsBootstrapper : Context refreshed
2023-07-03 11:55:08.150 - INFO  [           main] cumentationPluginsBootstrapper : Found 1 custom documentation plugin(s)
2023-07-03 11:55:08.185 - INFO  [           main] w.s.ApiListingReferenceScanner : Scanning for api listing references
2023-07-03 11:55:08.370 - INFO  [           main] .CachingOperationNameGenerator : Generating unique operation named: addUsingPOST_1
2023-07-03 11:55:08.455 - INFO  [           main] ledAnnotationBeanPostProcessor : No TaskScheduler/ScheduledExecutorService bean found for scheduled processing
2023-07-03 11:55:08.464 - INFO  [           main] o.s.b.StartupInfoLogger        : Started EmployeeApplication in 4.594 seconds (JVM running for 7.078)
2023-07-03 11:55:34.647 - INFO  [nio-8080-exec-1] o.a.j.l.DirectJDKLog           : Initializing Spring DispatcherServlet 'dispatcherServlet'
2023-07-03 11:55:34.647 - INFO  [nio-8080-exec-1] o.s.w.s.FrameworkServlet       : Initializing Servlet 'dispatcherServlet'
2023-07-03 11:55:34.648 - INFO  [nio-8080-exec-1] o.s.w.s.FrameworkServlet       : Completed initialization in 1 ms
2023-07-03 11:55:51.085 - INFO  [nio-8080-exec-8] c.z.h.HikariDataSource         : HikariPool-1 - Starting...
2023-07-03 11:55:51.244 - INFO  [nio-8080-exec-8] c.z.h.HikariDataSource         : HikariPool-1 - Start completed.
2023-07-03 11:55:51.314 - INFO  [nio-8080-exec-8] c.e.e.c.EmployeeController     : successfully insert an employee:tony
2023-07-03 13:35:56.873 - WARN  [l-1 housekeeper] c.z.h.p.HikariPool$HouseKeeper : HikariPool-1 - Thread starvation or clock leap detected (housekeeper delta=47m4s792ms432µs400ns).
2023-07-03 13:37:44.879 - INFO  [ioEventLoop-4-2] n.u.i.l.AbstractInternalLogger : null Unexpected exception during request: java.io.IOException: 远程主机强迫关闭了一个现有的连接。
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method) ~[?:1.8.0_342]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[?:1.8.0_342]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_342]
	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[?:1.8.0_342]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379) ~[?:1.8.0_342]
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:256) ~[netty-buffer-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1132) ~[netty-buffer-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:357) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.94.Final.jar:4.1.94.Final]
	at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_342]
2023-07-03 13:37:44.879 - INFO  [ioEventLoop-4-1] n.u.i.l.AbstractInternalLogger : null Unexpected exception during request: java.io.IOException: 远程主机强迫关闭了一个现有的连接。
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method) ~[?:1.8.0_342]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[?:1.8.0_342]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_342]
	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[?:1.8.0_342]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379) ~[?:1.8.0_342]
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:256) ~[netty-buffer-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1132) ~[netty-buffer-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:357) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.94.Final.jar:4.1.94.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.94.Final.jar:4.1.94.Final]
	at java.lang.Thread.run(Thread.java:750) ~[?:1.8.0_342]
2023-07-03 13:37:44.934 - INFO  [ecutorLoop-1-11] n.u.i.l.AbstractInternalLogger : Reconnecting, last destination was localhost/127.0.0.1:6379
2023-07-03 13:37:44.934 - INFO  [ecutorLoop-1-10] n.u.i.l.AbstractInternalLogger : Reconnecting, last destination was localhost/127.0.0.1:6379
2023-07-03 13:37:46.981 - WARN  [ioEventLoop-4-4] n.u.i.l.AbstractInternalLogger : Cannot reconnect to [localhost:6379]: Connection refused: no further information: localhost/127.0.0.1:6379
2023-07-03 13:37:46.981 - WARN  [ioEventLoop-4-3] n.u.i.l.AbstractInternalLogger : Cannot reconnect to [localhost:6379]: Connection refused: no further information: localhost/127.0.0.1:6379
2023-07-03 13:37:51.223 - INFO  [xecutorLoop-1-4] n.u.i.l.AbstractInternalLogger : Reconnecting, last destination was localhost:6379
2023-07-03 13:37:51.223 - INFO  [xecutorLoop-1-5] n.u.i.l.AbstractInternalLogger : Reconnecting, last destination was localhost:6379
2023-07-03 13:37:52.751 - INFO  [oEventLoop-4-10] i.l.c.p.ReconnectionHandler    : Reconnected to localhost:6379
2023-07-03 13:37:52.753 - INFO  [ioEventLoop-4-9] i.l.c.p.ReconnectionHandler    : Reconnected to localhost:6379
2023-07-03 13:42:14.526 - INFO  [nio-8080-exec-1] c.e.e.c.EmployeeController     : successfully insert an employee:dear-tony
